### DISCUSSION TITLE
Would be fun to be able to enter träwels in the past

### DISCUSSION THREAD

**Discussion by tml1024** [Traewelling/traewelling#2042] [Ideas / Feature Requests]
I only found the service now, but it would be fun to enter some träwels in the past that I still have exact memory of. 

But yeah, I understand that train numbers and timetables are not necessarily the same any more, so probably impossible to implement for arbitrary dates in the past. But at least for earlier träwels during the current timetable period?

**Comment by jheubuch** [comment]
Well you can? Just select the date on which you were travelling and the timetable connections (without real-time data) should be displayed:

![image](https://user-images.githubusercontent.com/60344409/222427655-8bea1fe6-dbea-4f6a-900c-14822bdecab9.png)

**Comment by tml1024** [comment]
Ah! Thanks, I hadn't looked closely enough. Closing then.

**Comment by jeyemwey** [comment]
I think this need still pays into "manual train checkin" piggybank which is discussed semi-regularly. Because Deutsche Bahn only provides info for trains so long from the past, some old data might be missing, especially if the trip was done months and years ago.

I'd like to ping @MrKrisKrisu here, and maybe even reopen the issue.

**Comment by MrKrisKrisu** [comment]
HAFAS also knows trips in the past, but the data also ends at some point. From 2-3 months back it is already difficult to get data via DB-Rest.

Anyway, we had the idea that you can create trips manually - especially helpful for people who work in the transport sector and want to record their empty trips and cargo trains with Träwelling.

In PR #1391 I created a test form in the admin panel where we can create trips manually to check in. On the backend side this is already working. In the frontend, however, this still leads to errors in some places. (e.g. 500s because of missing polyline).

**Comment by ErikUden** [comment]
Adding trips manually would be extremely helpful! I'd love to propose the idea that you can select a recent / currently occurring ride and simply say that this happened in the past. For example:

On **July 7th 2022** I took the 

**RE 3437 Train **

10:51 Venezia Mestre 
12:44 Trieste Centrale

This isn't an uncommon train! As a matter of fact, just today the same ride occurred.


So, my idea is: even if the APIs by traveling services across Europe don't include rides from months back, it's very likely the same ride still exists today, so Träwelling could pull the ride from today or a few weeks ago and simply use default times, then allow the user to edit / add delays in case they remember!

This would be very helpful for speed, travel time, etc. statistics as well as painting the exact path traveled on the map accurately.

Travelynx has a feature which allows you to manually add rides, but they'll only connect the different stations as the crow flies (Luftlinie), so the data is quite imprecise.

I just discovered Träwelling recently and would love to add my rides from my trip through Europe last year! Thanks for your development, I will probably use this app for many years to come.

**Comment by Luna131313** [comment]
In my opinion adding trips manually would be really helpful. I tried to add my older trips using the api, but I wasn't able to, because I don't know the hafas ID anymore. Would it be possible to implement a checkin request that doesn't require the hafas id?

**Reply by MrKrisKrisu** [reply]
You can now create manual journeys if you have activated the experimental features in the settings.

See [here](https://help.traewelling.de/features/manual-trips/).
